src/distributions/rayon.rs

```diff
@@ -0,0 +1,212 @@
+///
+/// This `struct` is created by the [`par_sample_iter`] method on
+/// [`Distribution`]. See its documentation for more.
+///
+/// [`Distribution`]: trait.Distribution.html
+/// [`sample_iter`]: trait.Distribution.html#method.sample_iter
+#[cfg(feature = "rayon")]
+#[derive(Debug)]
+pub struct ParallelDistIter<'a, D: 'a, R, T> {
+    distr: &'a D,
+    rng: R,
+    amount: usize,
+    phantom: ::core::marker::PhantomData<T>,
+}
+
+impl<'a, D, R, T> ParallelDistIter<'a, D, R, T> {
+    pub fn new(distr: &'a D, rng: &mut R, amount: usize)
+        -> ParallelDistIter<'a, D, R, T>
+        where D: Distribution<T>,
+              R: Rng + SeedableRng,
+    {
+        ParallelDistIter {
+            distr,
+            rng: R::from_rng(rng).unwrap(),
+            amount,
+            phantom: ::core::marker::PhantomData,
+        }
+    }
+}
+
+#[cfg(feature = "rayon")]
+impl<'a, D, R, T> ParallelIterator for ParallelDistIter<'a, D, R, T>
+    where D: Distribution<T> + Send + Sync,
+          R: Rng + SeedableRng + Send,
+          T: Send,
+{
+    type Item = T;
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+        where C: UnindexedConsumer<Self::Item>
+    {
+        bridge(self, consumer)
+    }
+
+    fn opt_len(&self) -> Option<usize> {
+        Some(self.amount)
+    }
+}
+
+#[cfg(feature = "rayon")]
+impl<'a, D, R, T> IndexedParallelIterator for ParallelDistIter<'a, D, R, T>
+    where D: Distribution<T> + Send + Sync,
+          R: Rng + SeedableRng + Send,
+          T: Send,
+{
+    fn drive<C>(self, consumer: C) -> C::Result
+        where C: Consumer<Self::Item>
+    {
+        bridge(self, consumer)
+    }
+
+    fn len(&self) -> usize {
+        self.amount
+    }
+
+    fn with_producer<CB>(self, callback: CB) -> CB::Output
+        where CB: ProducerCallback<Self::Item>
+    {
+        callback.callback(
+            DistProducer {
+                distr: self.distr.clone(),
+                amount: self.amount,
+                rng: self.rng,
+                phantom: ::core::marker::PhantomData,
+            }
+        )
+    }
+}
+
+/// FIXME
+#[cfg(feature = "rayon")]
+#[derive(Debug)]
+pub struct DistProducer<'a, D: 'a, R, T> {
+    distr: &'a D,
+    rng: R,
+    amount: usize,
+    phantom: ::core::marker::PhantomData<T>,
+}
+
+/// This method is intented to be used by fast and relatively simple PRNGs used
+/// for simulations etc. While it will also work with cryptographic RNGs, that
+/// is not optimal.
+///
+/// Every time `rayon` splits the work in two to create parallel tasks, one new
+/// PRNG is created. The original PRNG is used to seed the new one using
+/// `SeedableRng::from_rng`. **Important**: Not all RNG algorithms support this!
+/// Notably the low-quality plain Xorshift, the current default for `SmallRng`,
+/// will simply clone itself using this method instead of seeding the split off
+/// RNG well. Consider using something like PCG or Xoroshiro128+.
```

Comment:
I think it should be possible to use different RNGs for the seeding and the sampling.